poller-lambdas/src/aws.ts

```diff
@@ -0,0 +1,43 @@
+import { SecretsManagerClient } from '@aws-sdk/client-secrets-manager';
+import { SendMessageCommand, SQSClient } from '@aws-sdk/client-sqs';
+import { POLLER_LAMBDA_ENV_VAR_KEYS } from '../../shared/pollers';
+import {
+	getEnvironmentVariableOrCrash,
+	isRunningLocally,
+	remoteAwsConfig,
+} from './config';
+
+const buildLocalFakeSqsClient = () => {
+	const queueData: Record<string, SendMessageCommand[]> = {};
+	return {
+		queueData: queueData,
+		send: (command: SendMessageCommand) => {
+			const currentQueue: SendMessageCommand[] =
+				queueData[command.input.QueueUrl!] ?? [];
+			queueData[command.input.QueueUrl!] = [...currentQueue, command];
+			return Promise.resolve();
+		},
+	};
+};
+
+export const sqs = isRunningLocally
+	? buildLocalFakeSqsClient()
+	: new SQSClient(remoteAwsConfig);
+
+export const secretsManager = new SecretsManagerClient(remoteAwsConfig);
+
+const lambdaApp = process.env['App'];
+
+export const queueNextInvocation = (props: {
+	MessageBody: string;
+	DelaySeconds?: number;
+}) =>
+	sqs.send(
+		new SendMessageCommand({
+			QueueUrl: getEnvironmentVariableOrCrash(
+				POLLER_LAMBDA_ENV_VAR_KEYS.OWN_QUEUE_URL,
+			),
+			MessageDeduplicationId: lambdaApp, // should prevent the same lambda from being invoked multiple times
```

Comment:
do we still want this if the queue is no longer fifo?

Reply:
good spot, thanks, removed in https://github.com/guardian/newswires/pull/83/commits/910b766ed25cafee884e7760732392b7fe52c1b3